MaxText/layers/linears.py

```diff
@@ -371,7 +378,11 @@ def permute(self, inputs, gate_logits):
     inputs_shape = inputs.shape
     inputs_2d = jnp.reshape(inputs, (inputs_shape[0] * inputs_shape[1], inputs_shape[2]))
     weights, selected_experts = jax.lax.top_k(gate_logits, self.num_experts_per_tok)
-    weights = jax.nn.softmax(weights.astype(jnp.float32), axis=-1).astype(self.dtype)
+    if self.config.decoder_block == "deepseek":
```

Comment:
Can you add some comment how/why it's different for deepseek?

also, perhaps you can move it to a function, as it's used in 2 places, e.g.

def _deepseek_scale_weights(self, weights):
  """Scales weights according to DeepSeek's ... ."""
  weights /= weights.sum(-1, keepdims=True)
  weights *= self.config.routed_scaling_factor
  return weights

Reply:
SG, let me wrap it into a helper function.